test/YieldDirectorV2.test.js

```diff
@@ -0,0 +1,817 @@
+        // Deposit 1 gOHM into Tyche and donate to Bob
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+
+        // Verify donor info
+        const donationInfo = await tyche.depositInfo("0");
+        await expect(donationInfo.depositor).is.equal(deployer.address);
+        await expect(donationInfo.principalAmount).is.equal(`10${e9}`);
+        await expect(donationInfo.agnosticAmount).is.equal(principal);
+    });
+
+    it("should deposit sOHM tokens to recipient correctly", async () => {
+        // Deposit 10 sOHM into Tyche and donate to Bob
+        const principal = `10${e9}`;
+        await tyche.depositSohm(principal, bob.address);
+
+        // Verify donor info
+        const donationInfo = await tyche.depositInfo("0");
+        await expect(donationInfo.depositor).is.equal(deployer.address);
+        await expect(donationInfo.principalAmount).is.equal(principal);
+        await expect(donationInfo.agnosticAmount).is.equal(`1${e18}`);
+    });
+
+    it("should add deposits", async () => {
+        const principal = `1${e18}`;
+        const sohmPrincipal = `1${e9}`;
+
+        await tyche.deposit(principal, bob.address);
+        const donationInfo = await tyche.depositInfo("0");
+        await expect(donationInfo.depositor).is.equal(deployer.address);
+        await expect(donationInfo.principalAmount).is.equal(`10${e9}`);
+        await expect(donationInfo.agnosticAmount).is.equal(principal);
+
+        await tyche.addToDeposit("0", principal);
+        const donationInfo1 = await tyche.depositInfo("0");
+        await expect(donationInfo1.depositor).is.equal(deployer.address);
+        await expect(donationInfo1.principalAmount).is.equal(`20${e9}`);
+        await expect(donationInfo1.agnosticAmount).is.equal(`2${e18}`);
+
+        await tyche.addToSohmDeposit("0", sohmPrincipal);
+        const donationInfo2 = await tyche.depositInfo("0");
+        await expect(donationInfo2.depositor).is.equal(deployer.address);
+        await expect(donationInfo2.principalAmount).is.equal(`21${e9}`);
+        await expect(donationInfo2.agnosticAmount).is.equal("2100000000000000000");
+
+        await triggerRebase();
+
+        await tyche.connect(bob).redeemAllYield();
+
+        const donationInfo3 = await tyche.depositInfo("0");
+        const withdrawableBalance = await gOhm.balanceTo(donationInfo3.principalAmount);
+        await tyche.withdrawPrincipal("0", withdrawableBalance);
+    });
+
+    it("can't access all deposits per recipient", async () => {
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+
+        // Since entries in depositInfo are siloed by depositor, funds can't be exploited like this
+        await tyche.connect(alice).deposit(BigNumber.from("1"), bob.address);
+
+        const balanceBefore = await gOhm.balanceOf(alice.address);
+
+        await tyche.connect(alice).withdrawPrincipal("1", principal);
```

Comment:
to attempt to steal all the funds the amount you should put in withdrawPrincipal should be 1${e18} right. Thats how the exploit is done to old contract.

Reply:
oh yea oops